api/s2n.h

```diff
@@ -687,6 +687,266 @@ extern int s2n_cert_get_utf8_string_from_extension_data_length(const uint8_t *ex
 S2N_API 
 extern int s2n_cert_get_utf8_string_from_extension_data(const uint8_t *extension_data, uint32_t extension_len, uint8_t *out_data, uint32_t *out_len);
 
+/* Pre-shared key (PSK) Hash Algorithm - RFC 8446 Section-2.2 */
+typedef enum {
+    S2N_PSK_HMAC_SHA256,
+    S2N_PSK_HMAC_SHA384,
+} s2n_psk_hmac;
+
+struct s2n_psk;
+
+/**
+ * Creates a new s2n external Pre-shared key (PSK) object with `S2N_PSK_HMAC_SHA256` as the default 
```

Comment:
This is s2n, what do we gain by prefixing everything with s2n? Isn't that assumed?

(Applies to most of the comments)
```suggestion
/**
 * Creates a new external pre-shared key (PSK) object with `S2N_PSK_HMAC_SHA256` as the default 
```